### PR TITLE
Refactoring for cleaner interfaces in spark

### PR DESCRIPTION
**Pull request by nkronenfeld**
Convenience classes and implicits for constructing logs from a spark session or context
Using apache logging to log the actual messages, so that it plays nicely with existing logging

### CODE REVIEW

**Comment by nkronenfeld** [comment]
BTW, you asked about API changes herein.  There is one: There is no longer a local Level class, it just uses org.apache.log4j.Level instead

File: src/main/scala/software/uncharted/splog/Receiver.scala [hunk 1]
```diff
@@ -17,22 +17,20 @@
...
+import java.util.concurrent.{ExecutorService, Executors}
 
 private[splog] class Receiver(
   port: Int,
-  dateFormat: String = "yy/MM/dd HH:mm:ss z",
-  out: PrintStream = Console.out,
   threads: Int = 4
 ) extends Runnable {
-  import Level.{Level, TRACE, DEBUG, INFO, WARN, ERROR, FATAL, OFF} // scalastyle:ignore
+  // scalastyle:ignore
```
Comment (nkronenfeld) [inline src/main/scala/software/uncharted/splog/Receiver.scala]: This line should be gone

**Comment by nkronenfeld** [comment]
Sorry about the reduced coverage... The only ways I have found to invoke SparkSessionLogger.getLogger or SparkContextLogger.getLogger on a worker that actually compile throw an error before they get to the call.

* You can't mock the LoggerFactory - it's an object, not a class
* You can't pull in a SparkContext from the driver - SparkContext isn't Serializable
* You can't create a new SparkContext on the worker (for obvious reasons)

I'm pretty convinced this just can't be done.   I guess, in theory, we could just eliminate the check, but that just seems wrong.  At the very least, I like having it in there in case I'm wrong, and because it documents the necessary conditions for getting a logger.

If you've got any other ideas how to do this, I'd be happy to hear them.

**Comment by Ghnuberath** [comment]
This all looks good. The README needs adjusting, though.

**Pull request merged**